src/clr-angular/data/datagrid/datagrid-column-toggle.ts

```diff
@@ -3,141 +3,109 @@
+          type="button">
+          <clr-icon shape="close" [attr.title]="commonStrings.close"></clr-icon>
         </button>
-        <div class="column-switch"
-             *clrPopoverOld="open; anchor: anchor; anchorPoint: anchorPoint; popoverPoint: popoverPoint">
-            <div class="switch-header">
-                <ng-container *ngIf="!title">Show Columns</ng-container>
-                <ng-content select="clr-dg-column-toggle-title"></ng-content>
-                <button
-                    class="btn btn-sm btn-link"
-                    (click)="toggleUI()"
-                    type="button">
-                    <clr-icon shape="close" [attr.title]="commonStrings.close"></clr-icon>
-                </button>
-            </div>
-            <ul class="switch-content list-unstyled">
-                <li *ngFor="let column of columns">
-                    <clr-checkbox-wrapper>
-                        <input clrCheckbox type="checkbox"
-                          [disabled]="column.lastVisibleColumn"
-                          [ngModel]="!column.hidden"
-                          (ngModelChange)="toggleColumn($event, column)">
-                        <label><ng-template [ngTemplateOutlet]="column.template"></ng-template></label>
-                    </clr-checkbox-wrapper>
-                </li>
-            </ul>
-            <div class="switch-footer" *ngIf="buttons.length > 0">
-                <ng-content select="clr-dg-column-toggle-button"></ng-content>
-            </div>
-            <div class="switch-footer" *ngIf="buttons.length === 0">
-                <div>
-                    <button
-                            class="btn btn-sm btn-link p6 text-uppercase"
-                            [disabled]="allColumnsVisible"
-                            (click)="selectAll()"
-                            type="button">Select All
-                    </button>
-                </div>
-            </div>
+      </div>
+      <ul class="switch-content list-unstyled">
+        <li *ngFor="let column of hideableColumns;">
+          <clr-checkbox-wrapper>
+            <input clrCheckbox type="checkbox"
+                   [disabled]="isVisibleAlone && !column.value.isHidden"
+                   [ngModel]="!column.value.isHidden"
+                   (ngModelChange)="toggleColumn(!$event, column)">
+            <label>
+              <ng-template [ngTemplateOutlet]="column.value.titleTemplateRef"></ng-template>
+            </label>
+          </clr-checkbox-wrapper>
+        </li>
+      </ul>
+      <div class="switch-footer" *ngIf="buttons.length > 0">
+        <ng-content select="clr-dg-column-toggle-button"></ng-content>
+      </div>
+      <div class="switch-footer" *ngIf="buttons.length === 0">
+        <div>
+          <button
+            class="btn btn-sm btn-link p6 text-uppercase"
+            [disabled]="allColumnsVisible"
+            (click)="selectAll()"
+            type="button">Select All
```

Comment:
This should use the commonStrings service, and then we wouldn't need the `clr-dg-column-toggle-button` at all.

Reply:
As per our discussion, using commonStrings would introduce a breaking change. As this is internal refactoring change, I think it's better to not have any breaking changes for now. But that's being said, we could use the commonStrings service for both toggle title and button in the future.